test/test_runner_battle.c

```diff
@@ -10,6 +10,16 @@
 #include "test_battle.h"
 #include "window.h"
 
+// These are here to avoid breaking IDEs.
+#undef TestRunner_Battle_RecordAbilityPopUp
+#undef TestRunner_Battle_RecordAnimation
+#undef TestRunner_Battle_RecordHP
+#undef TestRunner_Battle_RecordMessage
+#undef TestRunner_Battle_RecordStatus1
+#undef TestRunner_Battle_AfterLastTurn
+#undef TestRunner_Battle_CheckBattleRecordActionType
+#undef TestRunner_Battle_GetForcedAbility
```

Comment:
Did you check if doing `#define TESTING` at the top of this file (it might be better more future-proof to put it in `test.h`) works for fixing your IDE errors? I would guess it will, because it should prevent these from being defined in the first place.

Reply:
It doesn't on either because the macros are defined in `include/test_runner.h`.

Reply:
But they're only defined in that file if `TESTING` isn't defined, right? So as long as your IDE can see that `TESTING` is defined before `include/test_runner.h` is included, you shouldn't get errors? (At least, in theory :sweat_smile:)

Reply:
![imagen](https://github.com/rh-hideout/pokeemerald-expansion/assets/2904965/32deb8e3-244a-44d6-9568-32fd84e01b44)

Reply:
Oh, sorry, I see what you've done that's my fault for not communicating clearly—I was thinking you'd `#define` testing literally on the first line. Like, before `#include "global.h"`. Does that help?

Reply:
Nope :/
![imagen](https://github.com/rh-hideout/pokeemerald-expansion/assets/2904965/dc45f40f-5b95-414e-94cd-92e3995de3d1)

Reply:
Fair enough! VS Code making our code worse yet again! :sob:

Reply:
I'll add `#if defined(__INTELLISENSE__)` to avoid it messing with stuff.